src/server.py

```diff
@@ -0,0 +1,168 @@
+# encoding: utf-8
+"""Small localtcp/ip server to host connections from local client."""
+
+import socket
+import datetime
+import os
+from mimetypes import MimeTypes
+
+walk_dir = os.getcwd()
+if walk_dir.split('/')[-1] != 'src':  # code to make tox work
+    walk_dir += '/src'
+walk_dir += '/webroot'
+
+
+buffer_length = 1024
+address = ('127.0.0.1', 5000)
+
+responses = {
+    400: ('Bad Request', 'Bad request syntax or unsupported method'),
+    404: ('Not Found', 'Nothing matches the given URI'),
+    405: ('Method Not Allowed', 'Specified method is invalid for this resource.'),
+    500: ('Internal Server Error', 'Server got itself in trouble'),
+    505: ('HTTP Version Not Supported', 'Cannot fulfill request.')
+}
+
+
+def response_ok(body, content_type):
+    """Return a well formed HTTP 200 OK response."""
+    response = b'HTTP/1.1 200 OK\r\n'
+    response += b'Date: ' + str(datetime.datetime.now().strftime('%a %b %Y %X PST')).encode() + b'\r\n'
+    response += b'Connection: close\r\n'
+    response += b'Content-Type: ' + content_type + b'\r\n'
+    response += b'Content-Length: ' + str(len(body)).encode() + b'\r\n\r\n'
+    response += body + b'\r\n'
+    return response
+
+
+def response_error(err_code):
+    """Return a well formed Error response."""
+    response = 'HTTP/1.1 {0} {1}\r\n'
+    response += 'Date: ' + datetime.datetime.now().strftime('%a %b %Y %X PST') + '\r\n'
+    return ''.join(response).format(err_code, responses[err_code][0])
+
+
+def parse_request(message):  # pragma: no cover
+    """Validate that the request is well-formed if it is return the URI from the request."""
+    request_split = message.split()
+    if request_split[0] != 'GET':
+        raise ValueError(405)
+    elif 'HTTP/' not in request_split[2]:
+        raise ValueError(400)
+    elif '1.1' not in request_split[2]:
+        raise ValueError(505)
+    return request_split[1]
+
+
+def folder_contents_html(folder_path, files, folders):
+    """Given files and folders generate html."""
+    html = "<!DOCTYPE html><html><body>{}</body></html>"
+    atag = '<a href="{}">{}</a>'
+    files_and_folders = ''
+    for folder in folders:
+        files_and_folders += '<h4>' + atag.format(folder_path + '/' + folder, folder) + '</h4>'
+    for file in files:
```

Comment:
It'd be best to use a different variable name than `file` as that's a reserved word.